src/Altinn.App.Api/Controllers/DataTagsController.cs

```diff
@@ -179,4 +192,110 @@ [FromRoute] string tag
 
         return NoContent();
     }
+
+    /// <summary>
+    /// Sets a set of tags on a data element.
+    /// </summary>
+    /// <param name="org">The short name for the application owner.</param>
+    /// <param name="app">The name of the application.</param>
+    /// <param name="instanceOwnerPartyId">The party id of the owner of the instance.</param>
+    /// <param name="instanceGuid">The id of the instance.</param>
+    /// <param name="dataGuid">The id of the data element.</param>
+    /// <param name="setTagsRequest">The request body.</param>
+    /// <param name="ignoredValidators">comma separated string of validators to ignore.</param>
+    /// <param name="language">The currently active user language.</param>
+    [HttpPut]
+    [Authorize(Policy = AuthzConstants.POLICY_INSTANCE_WRITE)]
+    [ProducesResponseType(typeof(SetTagsResponse), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<SetTagsResponse>> SetTags(
+        [FromRoute] string org,
+        [FromRoute] string app,
+        [FromRoute] int instanceOwnerPartyId,
+        [FromRoute] Guid instanceGuid,
+        [FromRoute] Guid dataGuid,
+        [FromBody] SetTagsRequest? setTagsRequest,
+        [FromQuery] string? ignoredValidators = null,
+        [FromQuery] string? language = null
+    )
+    {
+        if (!ModelState.IsValid || setTagsRequest is null)
+        {
+            return BadRequest(ModelState);
+        }
+
+        var tags = setTagsRequest.Tags;
+
+        if (tags.Any(tag => string.IsNullOrWhiteSpace(tag) || !LettersRegex().IsMatch(tag)))
+        {
+            return BadRequest("Tags may only contain letters, '-' and '_'.");
+        }
+
+        Instance instance = await _instanceClient.GetInstance(app, org, instanceOwnerPartyId, instanceGuid);
+        if (instance is null)
+        {
+            return NotFound("Unable to find instance based on the given parameters.");
+        }
+
+        DataElement? dataElement = instance.Data.FirstOrDefault(it =>
+            it.Id.Equals(dataGuid.ToString(), StringComparison.Ordinal)
+        );
+
+        if (dataElement is null)
+        {
+            return NotFound("Unable to find data element based on the given parameters.");
+        }
+
+        // Clear existing tags
+        dataElement.Tags ??= [];
+        dataElement.Tags.Clear();
+
+        // Add new tags
+        var normalizedTags = tags.Distinct(StringComparer.Ordinal).ToList();
+        dataElement.Tags.AddRange(normalizedTags);
```

Comment:
Nit:
Would't this be the same as
```suggestion
        // Add new tags
        dataElement.Tags = tags.Distinct(StringComparer.Ordinal).ToList();
```

Reply:
Haha true, I think so. Good catch 😅